middleware/interceptor.go

```diff
@@ -0,0 +1,33 @@
+package middleware
+
+import (
+	"bytes"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/gol4ng/httpware"
+)
+
+func Interceptor(callbackBeforeFunc func(*ResponseWriterInterceptor, *http.Request), callbackAfterFunc func(*ResponseWriterInterceptor, *http.Request)) httpware.Middleware {
```

Comment:
Add some explanation doc 👍

Reply:
really nice rework, this approach was better in design
i suggest you to implement a ReadCloserReseter in order to load request.Body in memory only when you read it and then reset only when you read it to